### PR TITLE
KAFKA-13256 Fix possible NPE when ConfigDef renders RST or HTML and ConfigKey.documentation is unset/NULL

### PR DESCRIPTION
**Pull request by rk3rn3r**
closes https://issues.apache.org/jira/browse/KAFKA-13256

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by showuon — APPROVED** [review]
LGTM! Thanks for the patch!

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java [hunk 1]
```diff
@@ -1372,11 +1372,13 @@ public String toEnrichedRst() {
      */
     private void getConfigKeyRst(ConfigKey key, StringBuilder b) {
         b.append("``").append(key.name).append("``").append("\n");
-        for (String docLine : key.documentation.split("\n")) {
-            if (docLine.length() == 0) {
-                continue;
+        if (null != key.documentation) {
```
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: Nit: we typically put `null` on the right hand side in the Kafka codebase. Same below

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java [hunk 2]
```diff
@@ -1372,11 +1372,13 @@ public String toEnrichedRst() {
...
+            for (String docLine : key.documentation.split("\n")) {
+                if (docLine.length() == 0) {
+                    continue;
+                }
+                b.append("  ").append(docLine).append("\n\n");
```
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:1380]: It seems `toHTLM()` is not tested but there's an existing test for `toEnrichedRst()` in `ConfigDefTest`. Can we add a config entry with a `null` documentation there?

What about something like:
```
diff --git clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java
index f8239e5b7e..5a4655db14 100644
--- clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java
+++ clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java
@@ -531,9 +531,15 @@ public class ConfigDefTest {
                         "Group Two", 1, Width.NONE, "..", Collections.<String>emptyList())
                 .define("opt1.of.group2", Type.BOOLEAN, false, Importance.HIGH, "Doc doc doc doc doc.",
                         "Group Two", 0, Width.NONE, "..", Collections.singletonList("some.option"))
-                .define("poor.opt", Type.STRING, "foo", Importance.HIGH, "Doc doc doc doc.");
+                .define("poor.opt", Type.STRING, "foo", Importance.HIGH, "Doc doc doc doc.")
+                .define("null doc", Type.STRING, "foo", Importance.HIGH, null);

         final String expectedRst = "" +
+                "``null doc``\n" +
+                "  * Type: string\n" +
+                "  * Default: foo\n" +
+                "  * Importance: high\n" +
+                "\n" +
```

**Review by mimaison — COMMENTED** [review]
Thanks for the PR! I left a couple of minor suggestions

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java (hunk 1 above)
Comment (rk3rn3r) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: applied

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java (hunk 2 above)
Comment (rk3rn3r) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:1380]: Hello @mimaison! Thanks for your review!
I applied the changes and added some code for it to the test cases.
I also discovered an issue with the RST output by adding the test!

**Review by mimaison — APPROVED** [review]
LGTM, thanks @rk3rn3r for the contribution!

**Pull request merged**